go-app-ussd_mcgcc_rapidpro.js

```diff
@@ -980,6 +982,183 @@ go.app = function() {
             });
         });
 
+        self.add("state_supporter_new_msisdn", function(name) {
+            return new FreeText(name, {
+                question: $("Please reply with the new cellphone number " +
+                    "you would like to get messages, e.g. 0762564733."),
+                check: function(content) {
+                    if (!utils.is_valid_msisdn(content, "ZA")) {
+                        return (
+                            "Sorry, we don't recognise that as a cell number. " +
+                            "Please reply with the 10 digit cell number, e.g. 0762564733."
+                        );
+                    }
+                    if (utils.normalize_msisdn(content, "ZA") === "+27762564733") {
+                        return (
+                            "Please try again. Reply with your new cellphone number " +
+                            "as a 10-digit number."
+                        );
+                    }
+                },
+                next: "state_supporter_new_msisdn_display"
+            });
+        });
+        
+        self.add("state_supporter_new_msisdn_display", function(name) {
+            var new_cell = self.im.user.answers.state_supporter_new_msisdn;
+            return new MenuState(name, {
+                question: $("Thank you! Let's make sure we got it right." +
+                    "\n\nIs your new number {{new_cell}}?").context({
+                    new_cell: new_cell
+                }),
+                error: $("Please try again. Reply with the nr that matchs your answer, e.g. 1").context(),
+                accept_labels: true,
+                choices: [
+                    new Choice("state_supporter_change_msisdn_rapidpro", $("Yes")),
+                    new Choice("state_supporter_new_msisdn", $("No, I want to retype my number"))
+                ]
+            });
+        });
+              
+        /*self.add("state_supporter_new_msisdn_whatsapp_contact_check", function(name, opts) {
```

Comment:
Should this be commented out?

Reply:
Yeah, it's older than the one I currently have locally. I'll regenerate